File: Backend/campusgear/campusgear/src/main/java/edu/cit/abadinas/campusgear/dto/AuthResponse.java
package edu.cit.abadinas.campusgear.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {

    private UserDto user;
    private String accessToken;
    private String refreshToken;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class UserDto {
        private Long id;
        private String email;
        private String firstname;
        private String lastname;
        private String role;
    }
}

File: Mobile/CampusGear/app/src/main/java/edu/cit/abadinas/campusgear/feature/auth/AuthResponse.kt
package edu.cit.abadinas.campusgear.feature.auth

import com.google.gson.annotations.SerializedName

/** Authentication response from the backend. */
data class AuthResponse(
    @SerializedName("user") val user: UserDto?,
    @SerializedName("accessToken") val accessToken: String?,
    @SerializedName("refreshToken") val refreshToken: String?
)

/** User DTO nested inside AuthResponse. */
data class UserDto(
    @SerializedName("id") val id: Long?,
    @SerializedName("email") val email: String?,
    @SerializedName("firstname") val firstname: String?,
    @SerializedName("lastname") val lastname: String?,
    @SerializedName("role") val role: String?
)
